internal/provider/recommendations_datasource.go

```diff
@@ -1 +1,175 @@
 package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-grafana-adaptive-metrics/internal/client"
+	"github.com/hashicorp/terraform-provider-grafana-adaptive-metrics/internal/model"
+)
+
+type recommendationDatasource struct {
+	client *client.Client
+}
+
+var (
+	_ datasource.DataSource              = &recommendationDatasource{}
+	_ datasource.DataSourceWithConfigure = &recommendationDatasource{}
+)
+
+func newRecommendationDatasource() datasource.DataSource {
+	return &recommendationDatasource{}
+}
+
+func (r *recommendationDatasource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*client.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected datasource configure type",
+			fmt.Sprintf("Got %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.client = data
+}
+
+func (r *recommendationDatasource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = fmt.Sprintf("%s_recommendations", req.ProviderTypeName)
+}
+
+func (r *recommendationDatasource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"verbose": schema.BoolAttribute{
+				Optional:    true,
+				Description: "If true, the response will include additional information about the recommendation, such as the number of rules, queries, and dashboards that use the metric.",
+			},
+			"recommendations": schema.ListNestedAttribute{
+				Computed: true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"metric": schema.StringAttribute{
+							Computed:    true,
+							Description: "The name of the metric to be aggregated.",
+						},
+						"match_type": schema.StringAttribute{
+							Optional:    true,
```

Comment:
Feels like you don't need to specify `Optional` at all for Computed attributes, I might be completely wrong here

Reply:
Yeah, I didn't express this correctly. I meant for datasources. For resources, Optional and Computed means that you can either set it or it's set by the API

Reply:
:+1: Makes sense, I've removed it.